src/components/auth/login.js

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import axiosWithAuth from '../../utils/axiosWithAuth'
+import { NavLink } from "react-router-dom";
+
+import { connect } from 'react-redux';
+import {loginAction} from '../../store/actions/loginAction';
+import './login.scss';
+// import Groa from '../auth/Groa-logo-B2.png'
+import Groa2 from '../auth/Groa-logo-B2AA.png'
+
+class LoginPage extends React.Component {
+  
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      user: {
+        user_name: '',
+        password: ''
+      },
+      errors: {
+        user_name: '',
+        password: '' 
+      }
+      
+  };
+
+
+  }
+     
+  
+    handleChange = e => {
+      const { name, value} = e.target;
+      let errors = this.state.errors;
+      e.preventDefault();
+      this.setState({
+          user: {
+          ...this.state.user,
+          [e.target.name]: e.target.value
+          },
+      });
+      switch (name) {
+        case 'user_name' :
+          errors.user_name =
+        value.length < 5
+        ? 'must be a valid user name'
+        : '';
+        break;
+        case 'password':
+          errors.password =
+          value.length < 4
+          ? 'Password must be more than 4 characters long'
+          : '';
+          break;
+          default:
+            break;
+      }
+  
+      this.setState({errors, [name]: value}, ()=> {
```

Comment:
what's the purpose of this function?

Reply:
displays errors if user has not filled form and hits Login.